src/include/mallocMC/creationPolicies/Scatter.hpp

```diff
@@ -515,69 +519,76 @@ namespace mallocMC
                                             return res;
                                     }
-                                    else if(chunksize == 0)
+                                    else if(beforechunksize >= bytes && beforechunksize <= maxchunksize)
                                     {
-                                        // lets open up a new page
-                                        const uint32 beforechunksize = alpaka::atomicOp<alpaka::AtomicCas>(
-                                            acc,
-                                            (uint32*) &_ptes[selected_page].chunksize,
-                                            0u,
-                                            minAllocation);
-                                        if(beforechunksize == 0)
-                                        {
-                                            void* res = tryUsePage(acc, selected_page, minAllocation);
-                                            if(res != 0)
-                                                return res;
-                                        }
-                                        else if(beforechunksize >= bytes && beforechunksize <= maxchunksize)
-                                        {
-                                            // someone else aquired the page,
-                                            // but we can also use it
-                                            void* res = tryUsePage(acc, selected_page, beforechunksize);
-                                            if(res != 0)
-                                                return res;
-                                        }
+                                        // someone else aquired the page,
+                                        // but we can also use it
+                                        void* res = tryUsePage(acc, page_in_region, beforechunksize);
+                                        if(res != 0)
+                                            return res;
                                     }
-                                    selected_page = region_offset + ((selected_page + 1) % regionsize);
-                                } while(selected_page != start_page);
-                                // could not alloc in region, tell that
-                                if(regionfilllevel + 1 <= regionsize)
-                                    alpaka::atomicOp<alpaka::AtomicCas>(
-                                        acc,
-                                        (uint32*) (_regions + region),
-                                        regionfilllevel,
-                                        regionfilllevel + 1);
-                            }
-                            start_page += regionsize;
+                                }
+                                page_in_region = region_offset + ((page_in_region + 1) % regionsize);
+                            } while(global_page != page_in_region);
```

Comment:
```suggestion
                            } while(page_in_region != global_page);
```
I think this is more natural as `page_in_region` is the loop counter here